### PR TITLE
lite-apiserver reuse proxy

### PR DESCRIPTION
**Pull request by Beckham007**
Based on #39 , lite-apiserver reuse proxy

### CODE REVIEW

File: pkg/lite-apiserver/cert/cert_manager.go [hunk 1]
```diff
@@ -56,10 +56,13 @@ func (m *CertManager) Init() error {
 	return nil
 }
 
-func (m *CertManager) Load(name string) *http.Transport {
+func (m *CertManager) GetTransport(name string) *http.Transport {
+	if len(name) == 0 {
+		return m.defaultTr
+	}
 	t, e := m.transportMap[name]
```
Comment (dixudx) [inline pkg/lite-apiserver/cert/cert_manager.go:63]: s/e/ok

Comment (dixudx) [reply pkg/lite-apiserver/cert/cert_manager.go:63]:
```go
if trans, ok := m.transportMap[name]; ok {
   return trans
}
return m.defaultTr

```

File: pkg/lite-apiserver/proxy/handler.go [hunk 2]
```diff
@@ -45,28 +51,64 @@ type EdgeServerHandler struct {
...
+	h.initProxies()
+
+	return h.buildHandlerChain(h), nil
+}
+
+func (h *EdgeServerHandler) initProxies() {
+	h.defaultProxy = NewEdgeReverseProxy(h.certManager.DefaultTransport(), h.apiserverUrl, h.apiserverPort, h.timeout, h.storage, h.cacher)
+
+	for commonName, transport := range h.certManager.GetTransportMap() {
+		proxy := NewEdgeReverseProxy(transport, h.apiserverUrl, h.apiserverPort, h.timeout, h.storage, h.cacher)
```
Comment (dixudx) [inline pkg/lite-apiserver/proxy/handler.go:74]: add some leveled logs here.

File: pkg/lite-apiserver/proxy/handler.go [hunk 3]
```diff
@@ -45,28 +51,64 @@ type EdgeServerHandler struct {
...
+		}
+	}
+
+	reverseProxy := h.getEdgeReverseProxy(commonName)
+	reverseProxy.ServeHTTP(w, r)
+}
+
+func (h *EdgeServerHandler) getEdgeReverseProxy(commonName string) *EdgeReverseProxy {
+	if len(commonName) == 0 {
+		return h.defaultProxy
```
Comment (dixudx) [inline pkg/lite-apiserver/proxy/handler.go:106]: Better add leveled logs here.

Comment (dixudx) [reply pkg/lite-apiserver/proxy/handler.go:106]:
```go
if proxy, ok := h.reverseProxyMap[commonName]; ok {
    klog.V(4).Infof("xxxx", xx)
    return proxy
}
klog.V(4).Warningf("xxxx",xx)
return h.defaultProxy

File: pkg/lite-apiserver/proxy/handler.go [hunk 4]
```diff
@@ -45,28 +51,64 @@ type EdgeServerHandler struct {
...
+	}
+
+	h.initProxies()
+
+	return h.buildHandlerChain(h), nil
+}
+
+func (h *EdgeServerHandler) initProxies() {
+	h.defaultProxy = NewEdgeReverseProxy(h.certManager.DefaultTransport(), h.apiserverUrl, h.apiserverPort, h.timeout, h.storage, h.cacher)
+
```
Comment (dixudx) [inline pkg/lite-apiserver/proxy/handler.go:72]: And have 

```go
h.reverseProxyMap[""] = h.defaultProxy
```

File: pkg/lite-apiserver/proxy/proxy.go [hunk 5]
```diff
@@ -306,37 +283,58 @@ func (p *EdgeReverseProxy) writeCache(header http.Header, statusCode int, pipeRe
...
-		klog.Errorf("Write cache %s error: %v", p.key(), err)
+		klog.Errorf("Write cache %s error: %v", key, err)
 		return err
 	}
 
 	return nil
 }
 
-func getRequestProperties(r *http.Request) (isList bool, isWatch bool, needCache bool) {
+func getRequestVerb(r *http.Request) (isList bool, isWatch bool) {
```
Comment (dixudx) [inline pkg/lite-apiserver/proxy/proxy.go:298]: Beter define a new function to get request info. And use that in both `getRequestVerb` and `needCache`.

File: pkg/lite-apiserver/cert/cert_manager.go (hunk 1 above)
Comment (Beckham007) [reply pkg/lite-apiserver/cert/cert_manager.go:63]: fixd in #44 as that pr refactor transport and cert

File: pkg/lite-apiserver/proxy/handler.go (hunk 2 above)
Comment (Beckham007) [reply pkg/lite-apiserver/proxy/handler.go:74]: fixd in #44 as that pr refactor transport and cert

File: pkg/lite-apiserver/proxy/handler.go (hunk 3 above)
Comment (Beckham007) [reply pkg/lite-apiserver/proxy/handler.go:106]: fixd in #44 as that pr refactor transport and cert

**Review by dixudx — APPROVED** [review]
/lgtm

**Pull request merged**